diesel/src/sqlite/types/date_and_time/chrono.rs

```diff
@@ -361,4 +436,142 @@ mod tests {
         let query = select(datetime("9999-01-08 00:00:00.000000").eq(distant_future));
         assert!(query.get_result::<bool>(connection).unwrap());
     }
+
+    #[test]
+    fn insert_timestamptz_into_table_as_text() {
```

Comment:
The idea behind this and the below test was to use `diesel` in its "canonical" form -- that is to have typed native Rust objects like `test_insert_timestamptz_into_table_as_text` that I could then manipulate with all the safety that Rust gurantees.

But these two tests are now giving me grief, very likely because I am still learning about the internals of `diesel` and how exactly it's intended to be used. I know that we're mixing metaphors a bit in trying to store datetime (w/ TZ) as text but still I thought this would've worked 😅 

Anyway pushing up to see if someone else can see what's obviously wrong with either the code or how I'm trying to use `diesel`; will keep poking away at it on my own too

```
@ diesel [stevenchu]$ cargo test --no-default-features --features 'extras sqlite'
   Compiling libsqlite3-sys v0.23.2
   Compiling diesel v2.0.0 (/Users/stevenchu/personal_projects/rust/diesel/diesel)
error[E0277]: the trait bound `type_impls::date_and_time::chrono::chrono::DateTime<Utc>: FromSql<sql_types::Text, Sqlite>` is not satisfied
   --> diesel/src/sqlite/types/date_and_time/chrono.rs:468:14
    |
468 |             .get_result::<DateTime<Utc>>(conn)
    |              ^^^^^^^^^^ the trait `FromSql<sql_types::Text, Sqlite>` is not implemented for `type_impls::date_and_time::chrono::chrono::DateTime<Utc>`
    |
    = help: the following implementations were found:
              <type_impls::date_and_time::chrono::chrono::DateTime<Local> as FromSql<Timestamptz, Sqlite>>
              <type_impls::date_and_time::chrono::chrono::DateTime<Utc> as FromSql<Timestamptz, Sqlite>>
note: required because of the requirements on the impl of `deserialize::Queryable<sql_types::Text, Sqlite>` for `type_impls::date_and_time::chrono::chrono::DateTime<Utc>`
   --> diesel/src/type_impls/date_and_time.rs:40:14
    |
40  |     #[derive(FromSqlRow)]
    |              ^^^^^^^^^^
...
51  |     struct DateTimeProxy<Tz: TimeZone>(DateTime<Tz>);
    |                                        ^^^^^^^^^^^^
    = note: required because of the requirements on the impl of `FromSqlRow<sql_types::Text, Sqlite>` for `type_impls::date_and_time::chrono::chrono::DateTime<Utc>`
note: required because of the requirements on the impl of `CompatibleType<type_impls::date_and_time::chrono::chrono::DateTime<Utc>, Sqlite>` for `sql_types::Text`
   --> diesel/src/query_dsl/load_dsl.rs:205:21
    |
205 |     impl<ST, U, DB> CompatibleType<U, DB> for ST
    |                     ^^^^^^^^^^^^^^^^^^^^^     ^^
note: required because of the requirements on the impl of `LoadQuery<'_, _, type_impls::date_and_time::chrono::chrono::DateTime<Utc>>` for `select_statement::SelectStatement<from_clause::FromClause<test_insert_timestamptz_into_table_as_text::table>, select_clause::SelectClause<test_insert_timestamptz_into_table_as_text::columns::timestamp_with_tz>>`
   --> diesel/src/query_dsl/load_dsl.rs:58:30
    |
58  | impl<'query, Conn, T, U, DB> LoadQuery<'query, Conn, U> for T
    |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^     ^
    = note: this error originates in the derive macro `FromSqlRow` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: the method `gt` exists for struct `select_statement::SelectStatement<from_clause::FromClause<test_query_timestamptz_column_with_between::table>, select_clause::SelectClause<test_query_timestamptz_column_with_between::columns::timestamp_with_tz>>`, but its trait bounds were not satisfied
   --> diesel/src/sqlite/types/date_and_time/chrono.rs:507:14
    |
507 |               .gt(Utc.ymd(1970, 1, 1).and_hms_milli(0, 0, 0, 0))
    |                ^^ method cannot be called on `select_statement::SelectStatement<from_clause::FromClause<test_query_timestamptz_column_with_between::table>, select_clause::SelectClause<test_query_timestamptz_column_with_between::columns::timestamp_with_tz>>` due to unsatisfied trait bounds
    |
   ::: diesel/src/query_builder/select_statement/mod.rs:45:1
    |
45  | / pub struct SelectStatement<
46  | |     From,
47  | |     Select = DefaultSelectClause<From>,
48  | |     Distinct = NoDistinctClause,
...   |
64  | |     pub(crate) locking: Locking,
65  | | }
    | | -
    | | |
    | | method `gt` not found for this
    | | doesn't satisfy `_: Iterator`
    | |_doesn't satisfy `_: expression::Expression`
    |   doesn't satisfy `_: global_expression_methods::ExpressionMethods`
    |
    = note: the following trait bounds were not satisfied:
            `select_statement::SelectStatement<from_clause::FromClause<test_query_timestamptz_column_with_between::table>, select_clause::SelectClause<test_query_timestamptz_column_with_between::columns::timestamp_with_tz>>: expression::Expression`
            which is required by `select_statement::SelectStatement<from_clause::FromClause<test_query_timestamptz_column_with_between::table>, select_clause::SelectClause<test_query_timestamptz_column_with_between::columns::timestamp_with_tz>>: global_expression_methods::ExpressionMethods`
            `&select_statement::SelectStatement<from_clause::FromClause<test_query_timestamptz_column_with_between::table>, select_clause::SelectClause<test_query_timestamptz_column_with_between::columns::timestamp_with_tz>>: expression::Expression`
            which is required by `&select_statement::SelectStatement<from_clause::FromClause<test_query_timestamptz_column_with_between::table>, select_clause::SelectClause<test_query_timestamptz_column_with_between::columns::timestamp_with_tz>>: global_expression_methods::ExpressionMethods`
            `select_statement::SelectStatement<from_clause::FromClause<test_query_timestamptz_column_with_between::table>, select_clause::SelectClause<test_query_timestamptz_column_with_between::columns::timestamp_with_tz>>: expression::Expression`
            which is required by `&select_statement::SelectStatement<from_clause::FromClause<test_query_timestamptz_column_with_between::table>, select_clause::SelectClause<test_query_timestamptz_column_with_between::columns::timestamp_with_tz>>: global_expression_methods::ExpressionMethods`
            `&mut select_statement::SelectStatement<from_clause::FromClause<test_query_timestamptz_column_with_between::table>, select_clause::SelectClause<test_query_timestamptz_column_with_between::columns::timestamp_with_tz>>: expression::Expression`
            which is required by `&mut select_statement::SelectStatement<from_clause::FromClause<test_query_timestamptz_column_with_between::table>, select_clause::SelectClause<test_query_timestamptz_column_with_between::columns::timestamp_with_tz>>: global_expression_methods::ExpressionMethods`
            `select_statement::SelectStatement<from_clause::FromClause<test_query_timestamptz_column_with_between::table>, select_clause::SelectClause<test_query_timestamptz_column_with_between::columns::timestamp_with_tz>>: Iterator`
            which is required by `&mut select_statement::SelectStatement<from_clause::FromClause<test_query_timestamptz_column_with_between::table>, select_clause::SelectClause<test_query_timestamptz_column_with_between::columns::timestamp_with_tz>>: Iterator`

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0277, E0599.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `diesel`

To learn more, run the command again with --verbose.
```

Reply:
These errors occur because there is no matching `FromSql` impl for strings. That seems to be required because of a derive somewhere. (I've not looked into the details here).
You should be able to easily fix that by adding similar impls to that ones: https://github.com/diesel-rs/diesel/blob/32bdcfde350bdf82138d0a16ef8cedd1e55fd938/diesel/src/sqlite/types/date_and_time/mod.rs#L52-L71

Reply:
Okay I think I figured it out here... needed to change the code to:
a/ actually type the Rust `table` object with the `TimestamptzSqlite` type, even if the underlying storage is `TEXT`
b/ use the `sql::<_>()` helper (is it a helper??) method to, again, type the value in the Rust type system -- during insert

Going to push up a new commit now after rebasing (I vaguely recall seeing some fixes for the `bin/test` suite that were also previously causing me grief) on `master`)

```rust
#[test]
fn insert_timestamptz_into_table_as_text() {
    crate::table! {
        #[allow(unused_parens)]
        test_insert_timestamptz_into_table_as_text(id) {
            id -> Integer,
            timestamp_with_tz -> TimestamptzSqlite,
        }
    }
    let conn = &mut connection();
    crate::sql_query(
        "CREATE TABLE test_insert_timestamptz_into_table_as_text(id INTEGER PRIMARY KEY, timestamp_with_tz TEXT);",
    )
    .execute(conn)
    .unwrap();

    let time: DateTime<Utc> = Utc.ymd(1970, 1, 1).and_hms_milli(0, 0, 0, 0);

    crate::insert_into(test_insert_timestamptz_into_table_as_text::table)
        .values(vec![(
            test_insert_timestamptz_into_table_as_text::id.eq(1),
            test_insert_timestamptz_into_table_as_text::timestamp_with_tz.eq(sql::<
                TimestamptzSqlite,
            >(
                "'1970-01-01 00:00:00+00:00'",
            )),
        )])
        .execute(conn)
        .unwrap();

    let result = test_insert_timestamptz_into_table_as_text::table
        .select(test_insert_timestamptz_into_table_as_text::timestamp_with_tz)
        .get_result::<DateTime<Utc>>(conn)
        .unwrap();
    assert_eq!(result, time);
}
```